### PR TITLE
[CodeSec AG] Add Private Registries [RLP-93689]

### PR DESCRIPTION
**Pull request by anaghapamidi**
Summary of Changes:

Modified Files:
- book.yml
- get-started/connect-your-repositories/connect-your-repositories.adoc

New Files:
- get-started/connect-your-repositories/add-private-registries.adoc
- _graphics/private-reg.png
- _graphics/private-reg-1.png
- _graphics/private-reg-2.png
- _graphics/private-reg-3.png
- _graphics/private-reg-4.png
- _graphics/private-reg-5.png
- _graphics/private-reg-6.png

### CODE REVIEW

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
+:topic_type: task
+
+[.task]
+== Add Private Registries to Prisma Cloud Code Security
+
+Private Registries integration with Prisma Cloud console enables you to find an accurate dependency tree of the packages in your private registries. The accuracy into your registries, ensures you have better fix suggestions for potentially vulnerable CVE.
+On Prisma Cloud scans are run on  both Private registries and public registries giving you a more concise scan results and fix suggestions.
+To remediate vulnerabilities found on Private registries see results and fix suggestions on *Projects*. Alternatively, you can make an informed decision by seeing the scan results on *Supply Chain*, where the graph displays the dependency tree of the latest private integration.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
Integrating Private Registries with Prisma Cloud enables building an accurate dependency tree of the packages in your private registries. This ensures you have more accurate fix suggestions for vulnerabilities in indirect packages and visibility into which packages are available in your private registry. To remediate vulnerabilities found on Private registries see results and fix suggestions on Projects. Alternatively, you can make an informed decision by seeing the scan results on Supply Chain, where the graph displays the dependency tree based on the integration. Prisma Cloud can integrate with Artifactory as a private registry.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+[.procedure]
+
+. Verify prerequisites.
++
+For Private Registries integration on Prisma Cloud Code Security, you need permissions and must have Prisma Cloud IP addresses and hostname on an allow list.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
For the Private Registries integration on Prisma Cloud Code Security, ensure you have the necessary permissions and Prisma Cloud IP addresses and hostname on an allow list.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
++
+* Access to the Prisma Cloud console.
++
+Ensure  your private registries have access to the Prisma Cloud IP addresses and hostname. You need access to 44.237.183.44 and 35.82.113.182. For more information see https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-admin/get-started-with-prisma-cloud/enable-access-prisma-cloud-console.html#id7cb1c15c-a2fa-4072-%20b074-063158eeec08[enable access to the Prisma Cloud Console].
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]: Those IPs are stack specific

```suggestion
Ensure  your private registries have access to the Prisma Cloud IP addresses and hostname. For more information see https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-admin/get-started-with-prisma-cloud/enable-access-prisma-cloud-console.html#id7cb1c15c-a2fa-4072-%20b074-063158eeec08[enable access to the Prisma Cloud Console].
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
++
+* Permissions required to see scan results after integration.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]: We need to add that the user must have credentials for Artifactory with either read only (BC) or admin (Cider) permissions.

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
++
+On the console you can see the scan results of Private registries at two instances.
++
+* *Projects*: Using the read-only permissions to private registries, you can see scan results for  Software Composition Analysis (SCA) on the console.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
* *Projects*: You can see scan results for Software Composition Analysis (SCA) on the console based on the private registry's dependency trees.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+* *Supply Chain*: Using the admin permission to  private registry, you can see the dependency tree on Supply Chain and make an informed decision.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
* *Supply Chain*: You can see the dependency tree on Supply Chain and make an informed decision about vulnerabilities and package use.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
++
+image::private-reg-3.png[width=600]
+
+.. Select *Registries* that package managers use.
++
+image::private-reg-4.png[width=600]
++
+NOTE: NPM requires you to define your repository as Private. An undefined NPM is by default a public repository.
++
+Optionally, you can select *Mirror Registry* to mirror the default integration.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
NPM requires that you define if your repository is Private.
+
For Maven, you can select *Mirror Registry* if the repository is a mirror of another repository.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+.. Select *Add Package*.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
.. Select *Select Package Manager Settings*.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
++
+image::private-reg-5.png[width=600]
+
+.. Select *Done*.
++
+Access *Settings > Repositories* to see the latest list of Private Registries integrations.
++
+To remediate scan results for private registries.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]: What does this mean? I'd delete it

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
++
+Access *Code Security > Projects > Vulnerabilities* to see the scan results and remediate a misconfiguration.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
Access *Code Security > Projects > Vulnerabilities* to see the scan results and remediate a vulnerability.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
++
+NOTE: You can see scan results for both public and private registries on *Vulnerabilities*. Therefore, the fix suggestions for CVE will reference the type of registry enabling you to differentiate your security priorities.
++
+Additionally, you can access *Code Security > Supply Chain* to view the dependency tree of the private packages and remediate a misconfiguration.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
Additionally, you can access *Code Security > Supply Chain* to view the dependency tree of the private packages.
```

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+.. Select *Done*.
++
+Access *Settings > Repositories* to see the latest list of Private Registries integrations.
++
+To remediate scan results for private registries.
++
+Access *Code Security > Projects > Vulnerabilities* to see the scan results and remediate a misconfiguration.
++
+NOTE: You can see scan results for both public and private registries on *Vulnerabilities*. Therefore, the fix suggestions for CVE will reference the type of registry enabling you to differentiate your security priorities.
```
Comment (tsmithv11) [inline code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]:
```suggestion
NOTE: Once a private registry is integrated, that is the default source for all packages. If a private fix version of a package is not available, Prisma Cloud will suggest a public fix version.
```

**Review by tsmithv11 — CHANGES_REQUESTED** [review]

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc (hunk 8 above)
Comment (anaghapamidi) [reply code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]: @tsmithv11 has the UI changed for this step? If yes can I see it on app-stage?

File: code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc (hunk 9 above)
Comment (anaghapamidi) [reply code-security/admin_guide/get-started/connect-your-repositories/add-private-registries.adoc]: Which line?

**Comment by anaghapamidi** [comment]
Summary of changes:

Modified Files:

- get-started/connect-your-repositories/add-private-registries.adoc
Includes review suggestions from @tsmithv11 

Open questions:
- Change in UI for line 59.
- Clarification for line 67.

**Review by anaghapamidi — COMMENTED** [review]
[This commit](https://github.com/PaloAltoNetworks/prisma-cloud-docs/pull/1109/commits/7175bf6588328371bacacc500bcb79cf942586c4) includes all suggestions.

**Comment by anaghapamidi** [comment]
Summary of changes:

Modified files:
- get-started/connect-your-repositories/add-private-registries.adoc

New File:
- private-reg-7.png

Note: Open question resolved through Slack conversation with Taylor.

**Review by tsmithv11 — APPROVED** [review]

**Pull request merged**